dhall/src/Dhall/Repl.hs

```diff
@@ -75,6 +79,7 @@ data Env = Env
   , explain          :: Bool
   , characterSet     :: CharacterSet
   , _standardVersion :: StandardVersion
+  , outputHandle     :: Maybe System.IO.Handle
```

Comment:
Perhaps instead of storing a `Maybe System.IO.Handle` you can store a `Text -> IO ()`.  Then when you want to write something you set it to `Data.Text.IO.putStrLn` and when you want to output nothing you set it to `mempty`

Reply:
We also use `renderIO` to pretty print Dhall, which uses the handle.
I could add an extra rendering function similar to what you suggest, as part of the state, but the `Maybe Handle` seems lighter. What do you prefer?